modules/core/react/lib/InputProvider.tsx

```diff
@@ -170,9 +175,17 @@ export default class InputProvider extends React.Component<InputProviderProps, I
   provideIntent = this.props.provideIntent;
 
   componentDidMount() {
+    const {container} = this.props;
+    if (container) {
+      // bail early if input provider is already handling this
+      if (container.classList.contains(containerClass)) {
+        return;
+      }
+      container.classList.add(containerClass);
```

Comment:
This was to provide access to the div we were rendering. Since the user controls that now, we don't need to add a classname anymore. If we need to check whether an element has an input provider attached, we can check for the `data-whatinput` attr